src/controllers/engine/colormapper.cpp

```diff
@@ -31,18 +31,13 @@ double colorDistance(QRgb a, QRgb b) {
 
 QRgb ColorMapper::getNearestColor(QRgb desiredColor) {
     // If desired color is already in cache, use cache entry
-    const auto iCachedColor = m_cache.constFind(desiredColor);
-    if (iCachedColor != m_cache.constEnd()) {
-        const QRgb cachedColor = iCachedColor.value();
-        DEBUG_ASSERT(m_availableColors.contains(cachedColor));
-        if (kLogger.traceEnabled()) {
-            kLogger.trace()
-                    << "Found cached color"
-                    << cachedColor
-                    << "for"
-                    << desiredColor;
-        }
-        return cachedColor;
+    QMap<QRgb, QRgb>::const_iterator i = m_cache.constFind(desiredColor);
+    if (i != m_cache.constEnd()) {
+        DEBUG_ASSERT(m_availableColors.contains(i.key()));
+        kLogger.trace()
+                << "ColorMapper cache hit for" << desiredColor << ":"
+                << "Color =" << i.value();
+        return i.value();
```

Comment:
Messed up merge? This undoes the fix from @uklotzde in #2733 and throws a debug assertion for me.

Reply:
I copied and pasted this function from master in d596ded. Please test it again. If so, I think this is ready for merge.